test/rational.jl

```diff
@@ -745,3 +745,318 @@ end
     @test rationalize(Int64, nextfloat(0.1) * im; tol=0) == precise_next * im
     @test rationalize(0.1im; tol=eps(0.1)) == rationalize(0.1im)
 end
+
+naive_rational_to_bigfloat(x) = BigFloat(numerator(x))/BigFloat(denominator(x))
+
+function test_rational_to_float_systematic_with_naive_impl(::Type{F}, q, rm) where {F}
+    no_alloc_rat = let
+        f(::Rational{Bool}) = true
+        function f(x)
+            if (sizeof(Int) < 8) && (8 < sizeof(x))
+                # `div(::Int128, ::Int128)` is implemented via `BigInt`on 32-bit Julia
+                false
+            else
+                isbits(widen(x))
+            end
+        end
+        f
+    end
+    want_no_alloc = no_alloc_rat(q) && isbitstype(widen(F))
+    want_no_alloc && @test iszero(@allocated F(q, rm))
+    @test F(naive_rational_to_bigfloat(q), rm) == @inferred F(q, rm)
+end
+
+function test_rational_to_float_systematic_with_naive_impl(::Type{BigFloat}, q, rm, prec)
+    reference = BigFloat(naive_rational_to_bigfloat(q), rm, precision = prec)
+    @test precision(reference) == prec
+    t = let reference = reference, prec = prec
+        function(res)
+            @test res == reference
+            @test precision(res) == prec
+        end
+    end
+
+    t(BigFloat(q, rm, precision = prec))
+
+    let
+        res = setprecision(BigFloat, prec) do
+            BigFloat(q, rm)
+        end
+        t(res)
+    end
+
+    let
+        if rm isa RoundingMode
+            res = setrounding(BigFloat, rm) do
+                BigFloat(q, precision = prec)
+            end
+            t(res)
+        end
+    end
+end
+
+function test_rational_to_float_systematic_with_naive_impl(::Type{BigFloat}, q, rm)
+    for prec ∈ Base.OneTo(10)
+        test_rational_to_float_systematic_with_naive_impl(BigFloat, q, rm, prec)
+    end
+end
```

Comment:
The small precisions are an edge case, so I definitely want them tested. Maybe it would make sense to throw some bigger precisions in, too?

Reply:
Although I don't think it makes sense to go above 256 bits, I don't think we're likely to catch any bugs that wouldn't be caught for the smaller precisions.

Reply:
You are right, small precision is of course an important edge case! I think it would make sense to test higher precision as well though. Testing with more than 256 bits would for example help with isolating any accidental dependence on the global precision that is used (that should be 256 bits in general). Maybe it's enough to test only 256 and 512 bits though! And maybe 1:5` instead of `1:10`? To reduce the runtime!